tests/sentry/snuba/test_discover.py

```diff
@@ -438,8 +442,295 @@ def test_params_forward(self, mock_query):
         )
 
 
-class TimeseriesQueryTest(TestCase):
-    pass
+class TimeseriesQueryTest(SnubaTestCase, TestCase):
+    def setUp(self):
+        super(TimeseriesQueryTest, self).setUp()
+
+        self.day_ago = before_now(days=1).replace(hour=10, minute=0, second=0, microsecond=0)
+
+        self.store_event(
+            data={
+                "event_id": "a" * 32,
+                "message": "very bad",
+                "timestamp": iso_format(self.day_ago + timedelta(hours=1)),
+                "fingerprint": ["group1"],
+                "tags": {"important": "yes"},
+            },
+            project_id=self.project.id,
+        )
+        self.store_event(
+            data={
+                "event_id": "b" * 32,
+                "message": "oh my",
+                "timestamp": iso_format(self.day_ago + timedelta(hours=1, minutes=1)),
+                "fingerprint": ["group2"],
+                "tags": {"important": "no"},
+            },
+            project_id=self.project.id,
+        )
+        self.store_event(
+            data={
+                "event_id": "c" * 32,
+                "message": "very bad",
+                "timestamp": iso_format(self.day_ago + timedelta(hours=2, minutes=1)),
+                "fingerprint": ["group2"],
+                "tags": {"important": "yes"},
+            },
+            project_id=self.project.id,
+        )
+
+    def test_invalid_field_in_function(self):
+        with pytest.raises(InvalidSearchQuery):
+            discover.timeseries_query(
+                selected_columns=["min(transaction)"],
+                query="transaction:api.issue.delete",
+                params={"project_id": [self.project.id]},
+                rollup=1800,
+            )
+
+    def test_no_aggregations(self):
+        with pytest.raises(InvalidSearchQuery) as err:
+            discover.timeseries_query(
+                selected_columns=["transaction", "title"],
+                query="transaction:api.issue.delete",
+                params={"project_id": [self.project.id]},
+                rollup=1800,
+            )
+        assert "no aggregation" in six.text_type(err)
+
+    @pytest.mark.xfail(
+        reason="Field aliases only work on transactions which does not have timeseries support yet."
+    )
```

Comment:
"time" should be available on the Discover dataset, hopefully this test passes now